### PR TITLE
Add render interpolator

### PR DESCRIPTION
**Pull request by iRevive**
This PR closes https://github.com/valskalla/odin/issues/64.

I've added a custom string interpolator, that uses the `Render` type class for every argument. 

Since the name is a matter of discussion, I want to keep this PR as WIP.

### CODE REVIEW

File: core/src/main/scala/io/odin/meta/Render.scala [hunk 1]
```diff
@@ -5,12 +5,23 @@ import cats.Show
 /**
   * Type class that defines how message of type `M` got rendered into String
   */
+@scala.annotation.implicitNotFound("""
+ No Render found for type ${M}. Try to implement an implicit Render[${M}].
+ You can implement it in ${M} companion class.
```
Comment (sergeykolbasov) [inline core/src/main/scala/io/odin/meta/Render.scala]: I would word it a bit differently:

> A value of type ${M} you're trying to log is missing a renderer.
> Make sure you have an implicit Render[${M}] in scope, i.e. inside of companion object of ${M}

**Comment by sergeykolbasov** [comment]
@iRevive is there anything you still want to add since it's still marked as WIP? Otherwise, it looks good to me.

**Comment by iRevive** [comment]
@sergeykolbasov you can merge it. Thanks.

**Comment by sergeykolbasov** [comment]
Thanks!

**Pull request merged**